### PR TITLE
[frontend] Initial approach to dynamic update of Title and Language for pages

### PR DESCRIPTION
**Pull request by ParamConstructor**
### Proposed changes
This PR is to address Accessibility Issues of  Issues (#6606 and #6607)
* [Change page title when navigating across the platform #6606](https://github.com/OpenCTI-Platform/opencti/issues/6606)
* [Change <html lang="XX"> depending on the selected language #6607](https://github.com/OpenCTI-Platform/opencti/issues/6607)

It has been applied only to the Root (for Language), Dashboard page, and the pages under Analysis (i.e. Reports, Groupings, Notes, Malware Analysis, and External References) to demonstrate the approach. If accepted, a future PR can be used to update the other pages using the useHeader component.

If the user's profile has selected "automatic", the page language will be determined by the navigator/browser language setting (i.e. navigator.language ). 

### Update: June 20, 2024

Uses a hook, versus the React Helmet module in initial commits to address @richard-julien comment/request to simplify the approach.

### Related issues

None

### Checklist

- [X] I consider the submitted work as finished
- [X] I tested the code for its functionality
- [X] (N/A) I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [X] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Comment by richard-julien** [comment]
Im a bit worried about the library used, that seems pretty old and not maintains anymore
React helmet async, last release Oct 8, 2018
React helmet, last release Jun 8, 2020

Could we try a more standard approach with hook and useEffect?
One more remark for lang that i think can be setup in root loading?

**Comment by ParamConstructor** [comment]
@richard-julien  - https://www.npmjs.com/package/react-helmet-async - was released last month? React-Helmet was replaced by it (the aysc version) and deprecated (https://www.npmjs.com/package/react-helmet).

**Comment by ParamConstructor** [comment]
@richard-julien - Updated to use a hook and not a specific module. The language is set only in the Root.

File: opencti-platform/opencti-front/src/utils/hooks/useHeader.ts
Comment (SarahBocognano) [inline opencti-platform/opencti-front/src/utils/hooks/useHeader.ts:1]: Is it possible to change the name of the hook as it's not really relevant. Maybe we could go for useDocumentTitle ? Or something more specific to the use.

Comment (ParamConstructor) [reply opencti-platform/opencti-front/src/utils/hooks/useHeader.ts:1]: @SarahBocognano - Yes - we are not hard set on it.

**Review by SarahBocognano — APPROVED** [review]

**Pull request merged**